packages/venia-concept/src/components/App/errorNotifications.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react';
+import { arrayOf, object, func, shape, string } from 'prop-types';
+import classify from 'src/classify';
+import { Notification, NotificationStack } from 'src/components/Notifications';
+import defaultClasses from './errorNotifications.css';
+
+class ErrorNotifications extends Component {
+    static propTypes = {
+        classes: shape({
+            debuginfo: string
+        }).isRequired,
+        errors: arrayOf(
+            shape({
+                error: object.isRequired,
+                id: string.isRequired,
+                loc: string
+            })
+        ),
+        onDismissError: func.isRequired
+    };
+
+    componentDidMount() {
+        window.scrollTo(0, 0);
+    }
+
+    render() {
+        const { classes, onDismissError, errors } = this.props;
+        if (errors.length > 0) {
+            return (
+                <div className={classes.root}>
+                    <NotificationStack>
+                        {errors.map(({ error, id, loc }) => (
```

Comment:
Consider doing multiline JS operations outside of JSX to preserve the tree structure.

```es6
const errorElements = errors.map(() => { ... })

return (
  <div>
    {errorElements}
  </div>
)
```